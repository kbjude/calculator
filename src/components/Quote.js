import React, { useState, useEffect } from 'react';
import quotes from '../Quotes.json';

const Quote = () => {
  const randomNumber = Math.round(Math.random() * 1);
  const [quote, setQuote] = useState('Beautiful quotes');
  useEffect(() => {
    setQuote(quotes[randomNumber]);
  }, []);
  return (
    <>
      <div data-testid="quote">
        {' '}
        { quote }
        {' '}
      </div>
    </>
  );
};

export default Quote;
