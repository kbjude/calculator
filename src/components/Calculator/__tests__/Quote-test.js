
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Quote from '../../Quote';

it('Succesfully displays', () => {
  const quote = renderer.create(<BrowserRouter><Quote /></BrowserRouter>).toJSON();
  expect(quote).toMatchSnapshot();
});
