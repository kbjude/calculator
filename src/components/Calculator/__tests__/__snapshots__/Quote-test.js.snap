// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Succesfully displays 1`] = `
<div
  data-testid="quote"
>
   
  Beautiful quotes
   
</div>
`;
